File: linked_list_find/linked_list_find.py
# class Node:
#   def __init__(self, val):
#     self.val = val
#     self.next = None
​
# O(n) time and O(n) space
# def linked_list_find(head, target):
#   if head is None: 
#     return False
  
#   if head.val == target:
#     return True
  
#   return linked_list_find(head.next, target)
    
# O(n) time and O(n) space  
def linked_list_find(head, target):
    while head is not None:
      if head.val == target:
        return True
      head = head.next
      
    return False
​
​
​
​
​
​
​
​
​
​
​
​
​
​
​

File: Linked-List-I/043-linked-list-find/linked-list-find.js
// class Node {
//   constructor(val) {
//     this.val = val;
//     this.next = null;
//   }
// }

//O(n) time and O(1) space where n is the number of nodes in the list
const linkedListFind = (head, target) => {
  let current = head;

  while (current !== null) {
    if (current.val === target) return true;
    current = current.next;
  }

  return false;
};

module.exports = {
  linkedListFind,
};
